File: core/src/main/java/pcb/uwu/units/derived/electromagnetism/ElectricConductanceUnit.java
package pcb.uwu.units.derived.electromagnetism;

import pcb.uwu.core.CompositeUnit;
import pcb.uwu.core.UnitCounter;
import pcb.uwu.units.base.ElectricCurrentUnit;
import pcb.uwu.units.base.LengthUnit;
import pcb.uwu.units.base.MassUnit;
import pcb.uwu.units.base.TimeUnit;

public class ElectricConductanceUnit extends CompositeUnit {

	// region base constructor

	public ElectricConductanceUnit(MassUnit massUnit, LengthUnit lengthUnit, ElectricCurrentUnit electricCurrentUnit, TimeUnit timeUnit) {
		super(new UnitCounter()
				.major(timeUnit, 3)
				.major(electricCurrentUnit, 2)
				.minor(massUnit)
				.minor(lengthUnit, 2));
	}

	// endregion

	// region derived constructors

	public ElectricConductanceUnit(ElectricResistanceUnit electricResistanceUnit) {
		super(new UnitCounter()
				.minor(electricResistanceUnit.getUnitCounter()));
	}

	// endregion

}

File: dimension/src/main/java/pcb/uwu/unit/derived/electromagnetism/ElectricConductanceUnit.kt
package pcb.uwu.unit.derived.electromagnetism

import pcb.uwu.core.CompositeUnit
import pcb.uwu.core.UnitCounter
import pcb.uwu.unit.base.ElectricCurrentUnit
import pcb.uwu.unit.base.LengthUnit
import pcb.uwu.unit.base.MassUnit
import pcb.uwu.unit.base.TimeUnit

open class ElectricConductanceUnit : CompositeUnit
{
    constructor(massUnit: MassUnit,
                lengthUnit: LengthUnit,
                electricCurrentUnit: ElectricCurrentUnit,
                timeUnit: TimeUnit)
            : super(UnitCounter()
                        .major(timeUnit)
                        .major(timeUnit)
                        .major(timeUnit)
                        .major(electricCurrentUnit)
                        .major(electricCurrentUnit)
                        .minor(massUnit)
                        .minor(lengthUnit)
                        .minor(lengthUnit))

    constructor(electricResistanceUnit: ElectricResistanceUnit)
            : super(UnitCounter()
                        .minor(electricResistanceUnit))
}
